src/web/mjs/connectors/MyAnimeListManga.mjs

```diff
@@ -20,30 +33,51 @@ export default class MyAnimeListManga extends Connector {
     }
 
     async _getMangas() {
-        let request = new Request(new URL('/read/manga', this.url), this.requestOptions);
-        let data = await this.fetchDOM(request, 'tr.ranking-list td.title div.detail > a:first-of-type');
-        return data.map(element => {
-            return {
-                id: element.href.split('/').pop(),
-                title: element.text.trim()
-            };
-        });
+        let mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
     }
 
-    async _getChapters(manga) {
-        let request = new Request(new URL('/read/manga/detail/' + manga.id, this.url), this.requestOptions);
-        let data = await this.fetchDOM(request, 'table.top-ranking-table tr td div.detail a:first-of-type');
+    async _getMangasFromPage(page) {
+        const uri = new URL('/store/search?p=' + page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.items a.item');
         return data.map(element => {
             return {
                 id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.text.trim(),
-                language: ''
+                title: element.querySelector('span.title').textContent.trim()
             };
         });
     }
 
+    async _getChapters(manga) {
+        let request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.v-manga-store-purchase-bulk-button');
+        const jObject= JSON.parse(data[0].dataset['items']);
+        return jObject.filter(element => element.is_previewable || element.is_free || element.is_possessed)
+            .map(element => {
+                const isPossessed = !element.is_possessed ? false : element.is_possessed; //value can be null instead of false
+                const isFull = element.is_free || isPossessed;
+                let title = element.numbering_text.trim();
+                title += isFull ? '' : ' [Preview]';
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(isFull ? element.viewer_url : element.preview_url, this.url),
+                    title: title,
+                    language: ''
+                };
+            });
+    }
+
     async _getPages(chapter) {
-        let request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        //TODO : add support for light novels  : its modified Publus reader
+        if (chapter.id.includes('novel_viewer')) {
+            throw new Error('Light Novels are not supported :/');
```

Comment:
May return an empty array instead of throwing an error, or filter the chapter list in andvance to exclude anything containing `novel_viewer`